### DISCUSSION TITLE
SetCameraAltControl & SetCameraAltControl & SetCameraMode disappeared?

### DISCUSSION THREAD

**Discussion by ZeroOut** [raysan5/raylib#3016] [Q&A] [answered]
SetCameraMode(camera, CAMERA_FREE);
SetCameraAltControl(KEY_LEFT_SHIFT);
SetCameraPanControl(MOUSE_BUTTON_LEFT);

Those functions no longer exist, how to implement？

**Comment by raysan5** [comment, marked as answer]
Camera system was completely redesigned from scratch. Now there is a `UpdateCameraPro()` that supports custom inputs.
